layers/erspan2.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2018 Google, Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file in the root of the source
+// tree.
+
+package layers
+
+import (
+	"encoding/binary"
+	"net"
+	"time"
+
+	"github.com/google/gopacket"
+)
+
+// Header contains all of the fields found in an ERSPAN2 header
+// https://tools.ietf.org/html/draft-foschiano-erspan-03
+type Header struct {
+	IsTruncated                         bool
+	Version, CoS, TrunkEncap            uint8
+	VLANIdentifier, SessionID, Reserved uint16
+	Index                               uint32
+}
+
+const (
+	// GREProtocolType - The protocol type value in the preceeding GRE header
+	GREProtocolType = 0x88BE
+	// VersionObsolete - The obsolete value for the version field
+	VersionObsolete = 0x0
+	// Version - The current value for the version field
+	Version = 0x1
+)
+
+// NewHeader is a convenience "constructor" which sets common default values
+func NewHeader() *Header {
+	return &Header{
+		Version: Version,
+	}
+}
+
+// DecodeFromBytes decodes the given bytes into this layer.
+func (h *Header) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	h.Version = data[0] & 0xF0 >> 4
+	h.VLANIdentifier = binary.BigEndian.Uint16(data[:2]) & 0x0FFF
+	h.CoS = data[2] & 0xE0 >> 5
+	h.TrunkEncap = data[2] & 0x18 >> 3
+	h.IsTruncated = data[2]&0x4>>2 != 0
+	h.SessionID = binary.BigEndian.Uint16(data[2:4]) & 0x03FF
+	h.Reserved = binary.BigEndian.Uint16(data[4:6]) & 0xFFF0 >> 4
+	h.Index = binary.BigEndian.Uint32(data[4:8]) & 0x000FFFFF
+	return nil
+}
+
+// SerializeTo writes the serialized form of this layer into the
+// SerializationBuffer, implementing gopacket.SerializableLayer.
+// See the docs for gopacket.SerializableLayer for more info.
+func (h *Header) SerializeTo(b gopacket.SerializeBuffer, opts gopacket.SerializeOptions) error {
+	bytes, err := b.PrependBytes(8)
+	if err != nil {
+		return err
+	}
+
+	twoByteInt := uint16(h.Version&0xF)<<12 | h.VLANIdentifier&0x0FFF
+	binary.BigEndian.PutUint16(bytes, twoByteInt)
+
+	twoByteInt = uint16(h.CoS&0x7)<<13 | uint16(h.TrunkEncap&0x3)<<11 | h.SessionID&0x03FF
+	if h.IsTruncated {
+		twoByteInt |= 0x400
+	}
+	binary.BigEndian.PutUint16(bytes[2:], twoByteInt)
+
+	fourByteInt := uint32(h.Reserved&0x0FFF)<<20 | h.Index&0x000FFFFF
+	binary.BigEndian.PutUint32(bytes[4:], fourByteInt)
+	return nil
+}
+
+// Conn - A connection to a device to send packets within ERSPAN2 encapsulation
```

Comment:
I don't think this "Conn" struct + methods should be in a Layer. I guess you are using that in your app code to send packets, but a layer should stick to DecodeFromBytes and SerializeTo methods

Reply:
Yeah, had a feeling that would be the feedback but worth a shot!  I'll remove the connection portions.